app/controllers/proposals_controller.rb

```diff
@@ -63,14 +63,33 @@ def create
   end
 
   def update
-    @url = conference_program_proposal_path(@conference.short_title, params[:id])
-
-    if @event.update(event_params)
-      redirect_to conference_program_proposals_path(conference_id: @conference.short_title),
-                  notice: 'Proposal was successfully updated.'
-    else
-      flash[:error] = "Could not update proposal: #{@event.errors.full_messages.join(', ')}"
-      render action: 'edit'
+    respond_to do |format|
```

Comment:
I would prefer to have a dedicated action for this and not merging it into update. As it update implies that the owner of the proposal updates his proposal. And I even think that this will cause problems with the permissions as you do ``load_and_authorize :event`` in the before filters. Which means only users with permissions to update the proposal/event can star it, right?